addons/udes_delivery_control/models/picking_extra_fields.py

```diff
@@ -0,0 +1,147 @@
+
+    picking_id = fields.Many2one("stock.picking", required=True, index=True, copy=False)
+
+    # Unloading Fields
+    u_supplier_id = fields.Many2one(
+        "res.partner", string="Supplier", domain="[('supplier', '=', True)]"
+    )
+    u_user_id = fields.Many2one("res.users", string="User", help="Picking completed by user")
+    u_delivery_note = fields.Char("Delivery Note")
+    u_location_id = fields.Many2one("stock.location", string="Location")
+    u_vehicle_scheduled_date = fields.Datetime(
+        "Scheduled Date/Time", help="Date/Time when the vehicle is expected to arrive"
+    )
+    u_vehicle_arrival_date = fields.Datetime("Arrival Date/Time", help="Arrival time of Vehicle")
+    u_week = fields.Integer("Week", compute="_compute_u_week", store=True, default=0)
+    u_unloading_start_date = fields.Datetime("Date Start", help="Unloading start date/time")
+    u_unloading_end_date = fields.Datetime("Date End", help="Unloading completed date/time")
+    u_time_taken = fields.Float(
+        "Time Taken HH:MM",
+        compute="_compute_u_time_taken",
+        store=True,
+        readonly=True,
+        help="The amount of time taken to completeunloading in HH:MM format",
+    )
+    u_vehicle_type = fields.Char("Vehicle Type")
+    u_trailer_number = fields.Char("Trailer Number")
+    u_lane_number = fields.Char("Lane Number")
+    u_pallet_count = fields.Integer("Pallet Count", default=0)
+    u_stillage_count = fields.Integer("Stillage Count", default=0)
+    u_box_count = fields.Integer("Box Count", default=0)
+    u_is_planned = fields.Boolean(
+        "Planned", help="Indicates whether the door control was planned or not", default=False,
+    )
+    u_is_late = fields.Boolean(
+        "Late", help="Indicates whether the unloadingwas late if booking slot set", default=False,
+    )
+    u_is_fast_track = fields.Boolean(
+        "Fast Track", help="Indicates whether the unloadinghas been fast tracked", default=False,
+    )
+
+    # Back Loading Fields
+    u_is_backload = fields.Boolean(
+        string="Backload",
+        help="Indicate whether a backload is required. \
+            (e.g. to take back cages used for past deliveries)",
+        default=False,
+    )
+    u_backload_supplier = fields.Many2one(
+        "res.partner",
+        "Supplier",
+        domain="[('supplier', '=', True)]",
+        help="Supplier from the backload",
+    )
+    u_backload_pallet_count = fields.Integer(
+        "Pallet Count", help="Number of pallets backloaded", default=0,
+    )
+    u_backload_stillage_count = fields.Integer(
+        "stillage Count", help="Number of stillages backloaded", default=0,
+    )
+    u_backload_cover_count = fields.Integer(
+        "Cover Count", help="Number of covers backloaded", default=0,
+    )
+    u_backload_reject_count = fields.Integer(
+        "Reject Count",
+        help="Number of  pallets/stillages/covers rejected and not backloaded",
```

Comment:
Nit: Double space between of and pallets 👀

Reply:
done